projects/cm_mcu/FireFlyTask.c

```diff
@@ -426,7 +426,7 @@ static int write_ff_register(const char *name, uint8_t reg, uint16_t value, int
     // write to register. First word is reg address, then the data.
     // increment size to account for the register address
     if ( ! res ) {
-      res = apollo_i2c_ctl_reg_w(i2c_device, ff_i2c_addrs[ff].dev_addr, reg, size, (int)value);
+      res = apollo_i2c_ctl_reg_w(i2c_device, ff_i2c_addrs[ff].dev_addr, 1, (uint16_t)reg, size, (uint32_t)value);
```

Comment:
I don't think you need the cast here on reg; this gets promoted automatically since uint16_t is a wider type.

Reply:
fixed it.